common/flatpak-parental-controls.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	 See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Authors:
+ *       Philip Withnall <withnall@endlessm.com>
+ */
+
+#include "config.h"
+
+#include <glib.h>
+#include <gio/gio.h>
+#include <appstream-glib.h>
+#include <libeos-parental-controls/app-filter.h>
+
+#include "flatpak-parental-controls-private.h"
+
+/* Convert an #EpcAppFilterOarsValue to an #AsContentRatingValue. This is
+ * actually a trivial cast, since the types are defined the same; but throw in
+ * a static assertion to be sure. */
+static AsContentRatingValue
+convert_app_filter_oars_value (EpcAppFilterOarsValue filter_value)
+{
+  G_STATIC_ASSERT (AS_CONTENT_RATING_VALUE_LAST == EPC_APP_FILTER_OARS_VALUE_INTENSE + 1);
```

Comment:
I don’t see why that would be any better. Neither of the two types are defined in flatpak, so we can’t put the static assertion next to either of the `typedef`s. Since this function is the one which converts between them, I think it’s appropriate that the static assertion is here.

Since the assertion is static, it doesn’t matter whether it’s inside or outside a function — if it’s going to fail, it fails at compile time regardless of where it’s put.

Reply:
I’m interested in why you think it would be better outside the function. :-)

Reply:
I just think it's weird to have a static assert in a dynamic context.